### PR TITLE
feat: adding vet to CI workflow

### LINKED ISSUE #645: Add `vet` to existing CI Workflow
**Current Behavior**
As of now, the current workflow allows code that contains suspicious or poorly written constructs.

**Desired Behavior**
The CI workflow must examine each PR for bad constructs using [vet](https://golang.org/cmd/vet/)
Example of a suspicious construct : 
```go
if <expr> {
   return true
}
return false
```
This can be better written as : 
```go
return <expr>
```
[vet](https://golang.org/cmd/vet/) ensures that the codebase is neat and developers follow good practices. ;)

---
**Resources**
Helpful tools and reference information - https://golang.org/cmd/vet/

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #645

### CODE REVIEW

**Comment by kushthedude** [comment]
@leecalcote Please review & merge!

File: mesheryctl/cmd/root.go [hunk 1]
```diff
@@ -82,14 +82,12 @@ var rootCmd = &cobra.Command{
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
-	fmt.Println("\n")
```
Comment (Aisuko) [inline mesheryctl/cmd/root.go:85]: Would you mind showing us how the effect of the code would be removed? Is there may make a non-new line situation? @kushthedude

Comment (kushthedude) [reply mesheryctl/cmd/root.go:85]: It was generating two blank lines which were not needed, as cursor was already printing on next line.

File: handlers/events_streamer.go [hunk 2]
```diff
@@ -57,7 +57,8 @@ func (h *Handler) EventStreamHandler(w http.ResponseWriter, req *http.Request, _
 	// defer close(newAdaptersChan)
 
 	go func() {
-		for mClient := range newAdaptersChan {
+		for loopVar := range newAdaptersChan {
```
Comment (Aisuko) [inline handlers/events_streamer.go:60]: The change does not make sense to me.

Comment (Aisuko) [reply handlers/events_streamer.go:60]:
```
for mClient := range newAdaptersChan {
      log.Debug("received a new mesh client, listening for events")
      go func(mClient *meshes.MeshClient) {
	      listenForAdapterEvents(req.Context(), mClient, respChan, log)
	      _ = mClient.Close()
}(mClient)
```

Comment (Aisuko) [reply handlers/events_streamer.go:60]: @leecalcote Any idea?

**Comment by kushthedude** [comment]
> do-not-merge/hold

I think we don't have k8s bot added 😅

File: handlers/events_streamer.go (hunk 2 above)
Comment (kushthedude) [reply handlers/events_streamer.go:60]: > These codes below make more sense to me
> 
> ```
> for mClient := range newAdaptersChan {
>       log.Debug("received a new mesh client, listening for events")
>       go func(mClient *meshes.MeshClient) {
> 	      listenForAdapterEvents(req.Context(), mClient, respChan, log)
> 	      _ = mClient.Close()
> }(mClient)
> ```

I agree with you, this makes more sense. All I did above was a makeover or can be termed as a cheat to bypass the rule for loop variable.

Comment (kushthedude) [reply handlers/events_streamer.go]: @Aisuko Fixed, Thanks for the suggestions !

**Comment by kushthedude** [comment]
@leecalcote This can be merged now

File: handlers/events_streamer.go [hunk 3]
```diff
@@ -57,14 +57,17 @@ func (h *Handler) EventStreamHandler(w http.ResponseWriter, req *http.Request, _
...
 	go func() {
+
 		for mClient := range newAdaptersChan {
-			log.Debug("received a new mesh client, listening for events")
-			go func() {
-				listenForAdapterEvents(req.Context(), mClient, respChan, log)
-				_ = mClient.Close()
-			}()
+      		log.Debug("received a new mesh client, listening for events")
+      		go func(mClient *meshes.MeshClient) {
```
Comment (Aisuko) [inline handlers/events_streamer.go:63]: Is there any idea about how to test this function?

Comment (kushthedude) [reply handlers/events_streamer.go:63]: @Aisuko I didn't quite get you here. Are you asking about the functionality test of the function or how will write unit tests for this function ?

Comment (kushthedude) [reply handlers/events_streamer.go:58]: If you are concerned about the functionality, I have tested the function and output is same as before.

Comment (Aisuko) [reply handlers/events_streamer.go:58]: @kushthedude This would be awesome, good job. Thanks

**Comment by Aisuko** [comment]
LGTM

**Pull request merged**